docs/csharp/language-reference/keywords/async.md

```diff
@@ -65,7 +65,7 @@ An async method can have the following return types:
 
 - <xref:System.Threading.Tasks.Task>
 - <xref:System.Threading.Tasks.Task%601>
-- [void](../../../csharp/language-reference/keywords/void.md), which should only be used for event handlers.
+- [void](../../../csharp/language-reference/keywords/void.md), which should only be used for event handlers or only call it from the main thread.
```

Comment:
How about this as a different update:

```suggestion
- [void](../../../csharp/language-reference/keywords/void.md). `async void` methods are generally discouraged because callers cannot `await` those methods and must implement a different mechanism to report successful completion or error conditions.
```

Reply:
That's fine.  I'd suggest changing "are generally discouraged" to "are generally discouraged for code other than event handlers"

Reply:
updated. Thanks @stephentoub. 

@Youssef1313 If you commit this suggestion, I’ll :shipit: